lte/gateway/c/session_manager/SetMessageManagerHandler.cpp

```diff
@@ -128,11 +173,42 @@ void SetMessageManagerHandler::send_create_session(
     } else {
       MLOG(MERROR)
           << "Failed to initialize 5G session for subscriber"
-          << cfg.common_context.sid().id() << " with PDU session ID "
+          << cfg.common_context.sid().id() << " with PDU session ID  from UE"
           << cfg.rat_specific_context.m5gsm_session_context().pdu_session_id()
           << " due to failure writing to SessionStore.";
     }
   }
 }
 
+/* This starts releasing the session in main session enforcer thread context
+ * Before startting it checks if respective session */
+void SetMessageManagerHandler::initiate_release_session(
+    SessionMap& session_map, const std::string& dnn, const std::string& imsi) {
+  // TODO as modification and dynamic rules are not implemented this may
+  // return empty map.
+  auto update = SessionStore::get_default_session_update(session_map);
+  bool exist =
+      m5g_enforcer_->m5g_release_session(session_map, imsi, dnn, update);
+  // If no entry found, nothing to do and return from here
+  if (!exist) {
+    MLOG(MERROR) << "Entry is not found in SessionStore for subscriber "
+                 << imsi;
+    return;
+  }
+
+  bool update_success = session_store_.update_sessions(update);
+  /* No need to respond AMF through gRPC as AMF has informed SessionD
+   * to release this session. if failed in update session & store
+   * print the fatal error and move on
+   */

```

Comment:
If this is just a matter of passing down the `response_callback` I would rather prefer that we respond with OK or error message

We don't want to promote situations where we are failing and we don't even see the error because it is hidden on some log

Reply:
Removed error log.